### PR TITLE
MetainfoScalarValue: fix UnicodeEncodeError

### PR DESCRIPTION
**Pull request by genestack-solomatin**
https://genestack.atlassian.net/browse/GS-11721

### CODE REVIEW

File: genestack_client/metainfo_scalar_values.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def _xstr(arg):
         :return: string representation of the object
         :rtype: str
         """
-        return str(arg) if arg is not None else None
+        return unicode(arg).encode('utf-8', 'ignore') if arg is not None else None
```
Comment (klas-genestack) [inline genestack_client/metainfo_scalar_values.py]: It's not nearly robust enough:
```python
>>> r'ы'
'\xd1\x8b'
>>> unicode(_)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    unicode(_)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd1 in position 0: ordinal not in range(128)
```

File: genestack_client/metainfo_scalar_values.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def _xstr(arg):
         :return: string representation of the object
         :rtype: str
         """
-        return str(arg) if arg is not None else None
+        return unicode(arg, 'utf-8').encode('utf-8', 'ignore') if arg is not None else None
```
Comment (klas-genestack) [inline genestack_client/metainfo_scalar_values.py]: Now this method accepts *only* `str` arguments.  If it is intended, docstring should be fixed (probably with exception handling).  But I personally would like to see a more thorough attempt to "cast" `object` to `str`.

File: ChangeLog [hunk 3]
```diff
@@ -4,6 +4,7 @@
     * Fixed error when response log printed without traceback.
     * Added `MetainfoRelatedValueFilter` file filter
     * Addded method `FilesUtil.find_metainfo_related_terms` for loading related terms
+    * Fixed `UnicodeEncodeError` when deserialize `MetainfoScalarValue`
```
Comment (klas-genestack) [inline ChangeLog]:
```
"Fix deserialization of Unicode `MetainfoScalarValue`s"
```

File: genestack_client/metainfo_scalar_values.py (hunk 2 above)
Comment (klas-genestack) [reply genestack_client/metainfo_scalar_values.py]: I've given it a bit more of thought: your changes are basically a no-op.  If `arg = None`, then you return `None`.  If it's a `str`, you _assume_ it's UTF8-encoded, decode it to `unicode`, then encode back to UTF8.  What's the point?

Comment (genestack-solomatin) [reply genestack_client/metainfo_scalar_values.py]: I see you point. Updated.

**Comment by klas-genestack** [comment]
This change has missed 0.29.0.

**Pull request merged**